protocol/communities_messenger_helpers_test.go

```diff
@@ -122,6 +126,78 @@ func (c *CollectiblesServiceMock) DeploymentSignatureDigest(chainID uint64, addr
 	return gethcommon.Hex2Bytes("ccbb375343347491706cf4b43796f7b96ccc89c9e191a8b78679daeba1684ec7"), nil
 }
 
+func findFreeUDPPort() (int, error) {
```

Comment:
Hmmm, I see some problems with this.

1. Range of TCP and UDP ports are different. A free TCP port can be busy for UDP.
2. I don't think it makes sense to have an "attempts loop". There're thousands of ports available locally, it's a very rare case if we don't have a free one.

### There's another function available

I found an function here which seems to be a better one:
https://github.com/status-im/status-go/blob/212e74fc1fe7aea31f736f3fbfe32686f73291df/wakuv2/waku.go#L160-L170

Though I think it's better to replace `net.IPv4zero` with `net.IP{127,0,0,1}`. Even though it's quite a fast operation, there's no need to make it publicly available, localhost is safer an more then enough.

### The approach 

Currently we go with this approach:
1. open a socket with port `0`
2. os gives us a free port
3. we remember it
4. we close the socket
5. we pass the remembered port to waku config
6. waku opens a udp socket on given port

I guess this will work fine in 99,999% of cases, but there's still a possibility that between clauses 4 and 6 the OS will give this port to some other socket. Then Waku will fail binding to this port.

I'm not sure what's the purpose of this port (I guess it's a discovery port?), probably it makes sense to have it fixed from the top level. But maybe for tests sake we could update waku library so that it assumes `port 0` as to select on automatically? And then if we need the port on top level, we just take it out with a method.

This would fix that 0.001% cases of problems. But what's more important, it will make the code clearer 🙂

Reply:
Wow, it's actually already done and even used by your code!. `getUsableUDPPort` is automatically called inside `waku.New(...)`:
https://github.com/status-im/status-go/blob/212e74fc1fe7aea31f736f3fbfe32686f73291df/wakuv2/waku.go#L188-L193

Reply:
Thank you, wow, i missed that func! So i guess it's fine to just rely on `getUsableUDPPort` inside `Waku.New`, correct? And leave `Config.UDPPort` at 0.

Reply:
yep, looks like that!
Though this should be done for the first instance only. For 2 other instances we should set the port as it's set in the first one.

Reply:
Sorry, don't understand - why should we set identical udp ports for different Waku instances?

Reply:
Removed the `findFreeUDPPort` func, relying on Waku's builtin func.

Reply:
> Sorry, don't understand - why should we set identical udp ports for different Waku instances?

Oh, never mind, I think I had a moment of misunderstanding 😄